src/org/mozilla/javascript/BaseFunction.java

```diff
@@ -105,16 +105,20 @@ protected int getMaxInstanceId() {
 
     @Override
     protected int findInstanceIdInfo(String s) {
-        int id;
+        int id = 0;
         switch (s) {
             case "length":
-                id = Id_length;
+                if (lengthPropertyAttributes >= 0) {
+                    id = Id_length;
+                }
```

Comment:
Because we have dropped the generator, we can think about replacing the whole ID assignment with direct return statements.

Reply:
Yes, we can delete a bunch of code now by doing that.